### ISSUE TITLE
fix disconnect between embedding evaluation levels

### ISSUE THREAD

**Issue by cmvcordova** [latent-reasoning-works/manylatents#64]
Dispatch currently specifies 3 levels:

```
@evaluate.register(dict)
def evaluate_embeddings(
    EmbeddingOutputs: dict,
    *,
    cfg: DictConfig,
    datamodule,
    **kwargs,
) -> dict:
    if EmbeddingOutputs is None or EmbeddingOutputs.get("embeddings") is None:
        logger.warning("No embeddings available for evaluation.")
        return {}
    
    if datamodule.mode == "split":
        ds = datamodule.test_dataset
    else:
        ds = datamodule.train_dataset ## defaults to full datset on full runs

    logger.info(f"Reference data shape: {ds.data.shape}")
    logger.info(f"Computing embedding metrics for {ds.data.shape[0]} samples.")
    
    metrics = {}
    all_metrics_cfg = cfg.metrics
    ds_metrics_cfg = all_metrics_cfg.get("dataset", {})
    ds_subsample_fraction = all_metrics_cfg.get("subsample_fraction", None)

    # naming and variable setting a bit fuzzy, make it be in line 
    # with module, embedding level metrics
    if ds_subsample_fraction is not None:
        ds_subsampled, embeddings_subsampled = subsample_data_and_dataset(ds, EmbeddingOutputs.get("embeddings"), ds_subsample_fraction)
        logger.info(f"Subsampled dataset to {embeddings_subsampled.shape[0]} samples for dataset-level metrics.")
    else:
        ds_subsampled, embeddings_subsampled = ds, EmbeddingOutputs.get("embeddings")

    module = kwargs.get("module", None)
    
    # dataset metrics
    for metric_name, metric_config in ds_metrics_cfg.items():
        metric_fn = hydra.utils.instantiate(metric_config)
        result = metric_fn(embeddings=embeddings_subsampled, dataset=ds_subsampled)
        metrics[metric_name] = result

    # embedding metrics
    embedding_metrics_cfg = all_metrics_cfg.get("embedding", {})
    for metric_name, metric_config in embedding_metrics_cfg.items():
        metric_fn = hydra.utils.instantiate(metric_config)
        # no subsetting here
        result = metric_fn(embeddings=EmbeddingOutputs.get("embeddings"),)
        metrics[metric_name] = result

    # module metrics
    module_metrics_cfg = all_metrics_cfg.get("module", {})
    for metric_name, metric_config in module_metrics_cfg.items():
        metric_fn = hydra.utils.instantiate(metric_config)
        # no subsetting here
        result = metric_fn(embeddings=EmbeddingOutputs.get("embeddings"), module=module)
        metrics[metric_name] = result

    return metrics

```

Which are then called with  to a top level config like this:
```

defaults:
- _self_
- dataset:
  - continuity
  - trustworthiness
- embedding:
  - _tangent_space_15

```

The disconnect happens at the logic level. The dispatch was thought of to deal with any logic specific to any of the 3 levels: dataset, module or embedding. If they required a specific evaluation routine, it would be handled there.

However, the metric protocol class got updated:

```
from src.algorithms.dimensionality_reduction import DimensionalityReductionModule

class Metric(Protocol):
    def __call__(self, 
                 embeddings: np.ndarray, 
                 dataset: Optional[object] = None, 
                 module: Optional[DimensionalityReductionModule] = None
            ) -> float:
        ...
    
    """A class that defines a metric for evaluating embeddings, irrespective of their modelling source.
    The metric is defined as a callable that takes embeddings as input and returns a float value.
    Optionally, a dataset and a dimensionality reduction module can be passed to the metric, to enable affinity and kernel matrix specific metrics.
    """
```

It now has embeddings as the "bare minimum" and the others as optional. Logic is handled within each metric, and should be, by design.

Proposed fix is currently to just update all metric definitions in the codebase to pass optional parameters as none, and have a single evaluation dispatch for embeddings that just passes the 3 as needed.

What must be checked:
See that base config paths don't break; these currently have sublevels and may have to be unrolled which may clutter the codebase too much. Specific configs will probably be purged from main, beyond the ones that serve for the basic partial instantiation call i.e. those currently in the `embedding/module/dataset` subfolders e.g. anisotropy or continuity in the embeddings folder.
